### DISCUSSION TITLE
In search of a decent lora board.

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#11577] [Hardware & Peripherals]
I was using LoPy4s for lora before Pycom went belly up. They were a pretty nice board, the socket.socket(socket.AF_LORA,socket.SOCK_RAW) interface just worked & I was getting 60mW (18 of the 20 dBm that Semtech promise) of RF.

I'm not having much luck finding a suitable replacement. The TTGO lora boards are only delivering 20mW (13 dbm) of RF & there doesn't seem to be much in the way of alternatives? Anybody know of a decent lora board with at least 40 mW (16 dBm) of output?

**Comment by davefes** [comment]
For the RF96W chip

> PA_HF and PA_LF are high efficiency amplifiers capable of yielding RF power programmable in 1 dB steps from -4 to
+14dBm directly into a 50 ohm load with low current consumption. PA_LF covers the lower bands (up to 525 MHz), whilst
PA_HF will cover the upper bands (from 860 MHz).

and

> PA_HP on PA_BOOST, any frequency +2 to +17dBm

and in sx127x.py
`PA_OUTPUT_PA_BOOST_PIN = 1`

```py

def set_tx_power(self, level, outputPin = PA_OUTPUT_PA_BOOST_PIN):
    self._tx_power_level = level
```
from that it looks like I am in boost-mode.  The 20dBm has a duty cycle associated with it.

**Reply by kjm1102** [reply]
Looks like the PA on is the default for the sx127x lib I'm using too & the 16 dBm I get with 'tx_power_level':17 in the config now is probably good enough.

**Comment by bidrohini** [comment]
I am not sure if you will find a LoRa board with 40mW output. However, if you consider a long range LoRa board instead, you can take a look at 
RN2903 LORA® TECHNOLOGY PICTAIL™/PICTAIL PLUS DAUGHTER BOARD 

https://www.microchip.com/en-us/development-tool/RN-2903-PICTAIL

RN2903 LORA is a good choice for IoT projects.

https://www.pcbway.com/project/shareproject/NP_Project_General_Purpose_IoT_Board_LoRa_WiFi_Bluetooth_RS485_ARM_C_462daacf.html

**Reply by kjm1102** [reply]
Thnx. I wish someone did something like https://www.sparkfun.com/products/18572 with a upython supported processor on the same pcb.

**Comment by mcauser** [comment]
Have you seen these two from eByte?

* [E22-900M30S](https://www.cdebyte.com/products/E22-900M30S) - Sx1262 30dBm SPI 12km
* [E32-900M30S](https://www.cdebyte.com/products/E32-900M30S) - Sx1276 30dBm SPI 10km

Sx126x is an upgrade from Sx127x.

The Sx126x give you more power/range than Sx127x using less current.

The SMD/SPI versions of the eByte boards let you use LoRaWAN.

Their UART 2.54mm header boards use LoRa modulation, but are point-to-point only (no WAN).
eg. [E32-900T20D](https://www.cdebyte.com/products/E32-900T20D) - Sx1276 20dBm UART 5.5km

**Reply by kjm1102** [reply]
That E22-900M30S seems nice. I wonder if the TCO allows bandwidths below 12.5kHz?

**Comment by dhalbert** [comment]
Note https://www.adafruit.com/product/5714, recently released. It is an RP2040 board with an SX127x.
